### PR TITLE
Add setting for number of processes to start

### LINKED ISSUE #2035: User control for number of processes
### Current Behaviour

MI will spawn a thread (`multiprocessing` process) for each CPU core on the machine. On high core count machines (especially on heterogeneous cpus in many new devices) this may not be the most effective.
`cores = context.cpu_count()` in `mantidimaging/core/parallel/manager.py`

### Desired Behaviour

Add a setting for the number of cores. Maybe a qsetting, that can be adjusted from the command line? Have a think about  #1890

Benchmark an operation (e.g. median (on CPU)) with a range of number of cores, on a range of hardware to see where the current scaling limit is. Evaluate if a low default would be sensible, e.g.
`cores = min(context.cpu_count(), 4)`. 

`operations_tests.py` might be useful for the benchmarking.

### PR DESCRIPTION
**Pull request by samtygier-stfc**
### Issue

Closes #2035 

### Description

Add a qsetting for number of processes to start in the multiprocessing pool. Lowering this can reduce startup time and memory usage. Previously MI would start a process of each CPU core detected, which can be too high in the case of hyper-threading and heterogeneous CPUs.

### Testing 

Make sure you have performance logging enabled in the qsettings. On linux in `.config/mantidproject/Mantid Imaging.conf` or in windows in the registry.
`logging/performance_log = true`

At startup you should see the number of processes started and the time to start the pool. In the qsetting set
`multiprocessing/process_count=4`
Start again an check there is a change

### Acceptance Criteria 

Startup should be slightly faster.
Operations should still work

### Documentation

Release notes

### CODE REVIEW

File: docs/release_notes/next/dev-2035-process-count [hunk 1]
```diff
@@ -0,0 +1 @@
+#2035: Add setting for number of processes to start in multiprocessing pool
```
Comment (JackEAllen) [inline docs/release_notes/next/dev-2035-process-count:1]: NITPICK: probably best to specify the number of processes we are capping startup too uin the release notes so it's easy to find if searching for this in the future.

**Review by JackEAllen — COMMENTED** [review]
Changes look good, I can confirm that results for linux, but not windows until i'm next on site. if anyone else can confirm on windows before then, feel free to take over this review. Otherwise, I'm happy with the changes - Just a small suggestion to modify the release notes to mention verbosely the number of processes to make it easier to find this PR should we ever need to in the future.

**Comment by MikeSullivan7** [comment]
Test on Windows, `process_count` in QSettings works as intended with slight improvement on startup.

```
(mantidimaging-dev) C:\Users\ddb29996\mantidimaging>python -m mantidimaging
2024-04-03 15:49:49,306 [mantidimaging.core.parallel.manager:L40] INFO: Creating process pool with 8 processes
2024-04-03 15:49:49,422 [perf.mantidimaging.core.parallel.manager:L44] INFO: Process pool started in 0.10900000005494803
2024-04-03 15:49:53,231 [perf.mantidimaging.gui.windows.main.view:L172] INFO: Mantid Imaging ready in 3.937999999965541
2024-04-03 15:49:53,231 [perf.mantidimaging.gui.mvp_base.view:L59] INFO: MainWindowView shown in 3.812999999965541

(mantidimaging-dev) C:\Users\ddb29996\mantidimaging>python -m mantidimaging
2024-04-03 15:50:57,822 [mantidimaging.core.parallel.manager:L40] INFO: Creating process pool with 4 processes
2024-04-03 15:50:57,876 [perf.mantidimaging.core.parallel.manager:L44] INFO: Process pool started in 0.0470000000204891
2024-04-03 15:51:00,780 [perf.mantidimaging.gui.windows.main.view:L172] INFO: Mantid Imaging ready in 2.969000000040978
2024-04-03 15:51:00,780 [perf.mantidimaging.gui.mvp_base.view:L59] INFO: MainWindowView shown in 2.9070000000065193
```

**Comment by samtygier-stfc** [comment]
> Once the screenshot tests pass on Windows, I'm happy to merge the PR

Not quite sure why that was failing, just a subtle positioning difference. I've updated the baselines.

**Review by MikeSullivan7 — APPROVED** [review]

**Pull request merged**